src/NHibernate/Async/Engine/BatchFetchQueue.cs

```diff
@@ -74,7 +74,7 @@ public Task<object[]> GetCollectionBatchAsync(ICollectionPersister collectionPer
 			foreach (KeyValuePair<CollectionEntry, IPersistentCollection> me in map)
 			{
 				cancellationToken.ThrowIfCancellationRequested();
-				if (await (ProcessKeyAsync(me)).ConfigureAwait(false))
+				if (await (ProcessKeyAndCheckCacheAsync(me)).ConfigureAwait(false))
```

Comment:
Awaiting for each cache key check also doesn't look effective... Though not sure yet how to improve it

Reply:
Inline is the answer :)